playground/backend/cmd/server/controller_test.go

```diff
@@ -372,148 +388,121 @@ func TestPlaygroundController_GetRunOutput(t *testing.T) {
 				if !strings.EqualFold(got.Output, tt.want.Output) {
 					t.Errorf("GetRunOutput() got = %v, want %v", got.Output, tt.want.Output)
 				}
-				if !reflect.DeepEqual(got.CompilationStatus, tt.want.CompilationStatus) {
-					t.Errorf("GetRunOutput() got = %v, want %v", got.CompilationStatus, tt.want.CompilationStatus)
-				}
 			}
 		})
 	}
 }
 
-func Test_processCode(t *testing.T) {
+func TestPlaygroundController_GetRunError(t *testing.T) {
+	ctx := context.Background()
 	pipelineId := uuid.New()
-	networkEnvs, err := environment.GetNetworkEnvsFromOsEnvs()
-	if err != nil {
-		panic(err)
-	}
-	appEnvs, err := environment.GetApplicationEnvsFromOsEnvs()
-	if err != nil {
-		panic(err)
-	}
-	sdkEnv, err := environment.ConfigureBeamEnvs(appEnvs.WorkingDir())
+	runError := "MOCK_RUN_ERROR"
+	conn, err := grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
 	if err != nil {
-		panic(err)
+		t.Fatalf("Failed to dial bufnet: %v", err)
 	}
-	env := environment.NewEnvironment(*networkEnvs, *sdkEnv, *appEnvs)
-
-	lc, _ := fs_tool.NewLifeCycle(pb.Sdk_SDK_JAVA, pipelineId, os.Getenv("APP_WORK_DIR"))
-	filePath := lc.GetAbsoluteExecutableFilePath()
-	workingDir := lc.GetAbsoluteExecutableFilesFolderPath()
-
-	exec := executors.NewExecutorBuilder().
-		WithValidator().
-		WithSdkValidators(validators.GetJavaValidators(filePath)).
-		WithCompiler().
-		WithCommand(sdkEnv.ExecutorConfig.CompileCmd).
-		WithArgs(sdkEnv.ExecutorConfig.CompileArgs).
-		WithFileName(filePath).
-		WithWorkingDir(workingDir)
+	defer conn.Close()
+	client := pb.NewPlaygroundServiceClient(conn)
 
 	type args struct {
-		ctx context.Context
-		env *environment.Environment
-		sdk pb.Sdk
+		ctx  context.Context
+		info *pb.GetRunErrorRequest
 	}
 	tests := []struct {
-		name                  string
-		createExecFile        bool
-		code                  string
-		expectedStatus        pb.Status
-		expectedRunOutput     interface{}
-		expectedCompileOutput interface{}
-		args                  args
+		name    string
+		prepare func()
+		args    args
+		want    *pb.GetRunErrorResponse
+		wantErr bool
 	}{
 		{
-			// Test case with calling processCode method without preparing files with code.
-			// As a result status into cache should be set as Status_STATUS_VALIDATION_ERROR.
```

Comment:
I find these comments for tests very useful to understand what's going on. Can we add them, please?

Reply:
Added comments for tests.

Reply:
Since it's a refactoring, could you please add comments for all tests in this file? That would help maintainers a lot😊